portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -126,4 +131,22 @@ private String getBody(HttpServletRequest request) throws IOException {
 
         return "";
     }
+
+    private double getSentiment(String comment) throws IOException {
+        Document document = Document
+            .newBuilder()
+            .setContent(comment)
+            .setType(Document.Type.PLAIN_TEXT)
+            .build();
+
+        LanguageServiceClient languageService = LanguageServiceClient.create();
```

Comment:
I'd make this a private variable on this class so you can re-use the language service on every POST instead of re-creating it. 

In this particular case, there's probably not much overhead to calling LanguageServiceClient.create(), but you could imagine if this was something like connect to database, it might be wasteful to re-connect every time.

Reply:
Yeah that's true. But I am not sure if it'd be possible to instantiate the service once, because as I've read, every time to finish using it, you have to close the service. But if it gets closed after the first request and then not instantiated again, it will not be available.

I am thinking probably of opening the service on the constructor, and closing it on the destructor to reduce that overhead that you're mentioning it. WDYT?

Reply:
Hmm, I see what you mean about needing to close. Ok, I see two equally reasonable options here.

1: Just ignore my suggestion, and just use the try-with-resources block style they have in the sample code to make it a little cleaner: https://cloud.google.com/natural-language/docs/reference/libraries#client-libraries-usage-java

2: Implement init/destroy of HttpServlet Class https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServlet.html and calling close() in the destroy. which is pretty much the same as your constructor/destructor suggestion but specific to HttpServlet.


Which one is better depends on how many comments you expect to get. If comments are very frequent, it's probably worth keeping a connection to the service around. If comments are fewer or more spread out, it's probably not that big of deal to just reconnect for each comment.

For a portfolio site, you can reasonably assume fewer comments, but maybe a site like Reddit would expect frequent enough comments to prefer keeping service connections alive.

Reply:
Yeah, sounds reasonable. I guess the amount of comments submitted does really impact the implementation. I think I will stick to the try-with-resources approach to clean up the code since it is true that for a webapp of this type we shouldn't be expecting massive interaction.

Thanks for the options though. It's nice to know the different approaches.